cachelib/navy/common/Device.cpp

```diff
@@ -863,6 +981,15 @@ FileDevice::FileDevice(std::vector<folly::File>&& fvec,
   // for the path where device IO is called from non-fiber thread
   // (e.g., recovery path, read random alloc path)
   syncIoContext_ = std::make_unique<SyncIoContext>();
+#ifndef CACHELIB_IOURING_DISABLE
+  if (isFDPEnabled_) {
+    fdpNvme_ = std::make_shared<FdpNvme>(fvec_[0].fd());
```

Comment:
How do we guarantee `fdpNvme_` is initialized successfully? If FDP is not enabled, we should reset it.

Reply:
it was initialized to nullptr.
std::shared_ptr<FdpNvme> fdpNvme_{nullptr};

Reply:
I meant, what if the constructor fails to initialize FDP NVMe.